tests/migration/sle12_online_migration/zypper_patch.pm

```diff
@@ -28,6 +28,7 @@ sub run {
     fully_patch_system;
     remove_ltss;
     power_action('reboot', keepconsole => 1, textmode => 1);
+    reconnect_mgmt_console if (check_var('ARCH', 'ppc64le'));
```

Comment:
+1 here, I guess you wanted to use `is_pvm` here too. power KVM doesn't need it.

Reply:
Yes, before I use is_pvm, But please check Oliver suggested 'try to not put any backend specific code into test modules'. So I change to check 'ARCH', at reconnect_mgmt_console, it will check is_pvm condition.

Reply:
The point is that with this condition we will match qemu backend too which is wrong. I'm with @okurz in general, but unfortunately we cannot fix the world in the single PR. So in case we cannot avoid `is_pvm` condition, we can address it later as we have such code in many other places.
If we can reuse generic `is_remote_backend` subroutine here, would be great, if not, leave `is_pvm` and let's file follow-up ticket to refactor code for rebooting on powerVM and other remote backends, so that we don't need any backend specific code in the test modules.

Reply:
I agree that `is_pvm` is better than `check_var('ARCH'…)`. Sorry that I could not make myself clear. When I meant "try to not put any backend specific code into test modules" then the same applies for ARCH as well :) 

What I suggested to you in general is also applicable here: Try to write the test code – especially in the test modules themselves – as much as possible from the point of view of a human tester. The code should state the intention. Any backend specific handling should at least not confuse the reader of the test module on the top level. At best the backend itself handles special behaviour but if we can not achieve that then at least we should use the helper functions in the lib.

Of course we can separate the work and do one step at a time however I think now (after 15SP2 GMC) is likely the best time to do it properly :)

Reply:
I will change back to use is_pvm. But I don't understand what do you mean 'at least we should use the helper functions in the lib'?